app/assets/javascripts/discourse/app/lib/post-action-feedback.js

```diff
@@ -0,0 +1,90 @@
+import { SVG_NAMESPACE } from "discourse-common/lib/icon-library";
+import I18n from "discourse-i18n";
+
+export default function postActionFeedback({
+  postId,
+  actionClass,
+  messageKey,
+  actionCallback,
+  errorCallback,
+}) {
+  if (recentlyCopied(postId, actionClass)) {
+    return;
+  }
+
+  const maybePromise = actionCallback();
+
+  if (maybePromise && maybePromise.then) {
+    maybePromise
+      .then(() => {
+        showAlert(postId, actionClass, messageKey);
+      })
+      .catch(() => {
+        if (errorCallback) {
+          errorCallback();
+        }
+      });
+  } else {
+    showAlert(postId, actionClass, messageKey);
+  }
+}
+
+function recentlyCopied(postId, actionClass) {
+  return document.querySelector(
+    `article[data-post-id='${postId}'] .${actionClass} .${actionClass}-checkmark`
+  );
+}
+
+export function showAlert(postId, actionClass, messageKey) {
```

Comment:
```suggestion 
  function showAlert(postId, actionClass, messageKey) {
```

This doesn't seem like it needs to be exported anymore

Reply:
it does, it is consumed by the widget